packages/@expo/cli/src/start/server/metro/withMetroMultiPlatform.ts

```diff
@@ -754,13 +754,15 @@ export function withExtendedResolver(
         if (context.customResolverOptions?.environment === 'react-server') {
           context.unstable_conditionNames = [
             'node',
-            'import',
-            'require',
+            context.isESMImport === true ? 'import' : 'require',
```

Comment:
Probably need same logic as this to respect the entire config https://github.com/facebook/metro/blob/c945f597049e3ae512f9d81862664951ec05f5b3/packages/metro-resolver/src/utils/matchSubpathFromExportsLike.js#L37-L44

Reply:
Actually just ensure this logic is in the custom resolver and delete both conditions here

Reply:
I moved `context.isESMImport` out of `withMetroMultiPlatform`, and solely rely on what's being set in the fast resolver or default resolver. I think the other conditions for `node` and `react-server` still need to be applied to both resolvers - and should not be moved to fast resolver only - right?